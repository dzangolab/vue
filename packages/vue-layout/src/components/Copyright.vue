<template>
  <div class="copyright">
    &copy; {{ year }} <a :href="url" target="_blank">{{ holder }}</a
    >&nbsp;
    {{ copyright }}
  </div>
</template>

<script lang="ts">
export default {
  name: "Copyright",
};
</script>

<script setup lang="ts">
import { useI18n } from "@prefabs.tech/vue3-i18n";
import { computed } from "vue";

import { useTranslations } from "../index";

import type { LocaleMessages, VueMessageType } from "@prefabs.tech/vue3-i18n";
import type { PropType } from "vue";

const props = defineProps({
  holder: {
    required: true,
    type: String as PropType<string>,
  },
  showDisclaimer: Boolean,
  url: {
    required: true,
    type: String as PropType<string>,
  },
});

const messages = useTranslations() as LocaleMessages<VueMessageType>;

const { t } = useI18n({ messages });

const copyright = computed(() => {
  return props.showDisclaimer ? t("copyright.disclaimer") : null;
});

const year = new Date().getFullYear();
</script>
