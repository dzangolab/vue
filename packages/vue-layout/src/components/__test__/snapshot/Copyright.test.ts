import i18nPlugin, { useLocaleStore } from "@dzangolab/vue3-i18n";
import { shallowMount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";

import Copyright from "@/components/Copyright.vue";

import appConfig from "../config";

import type { VueWrapper } from "@vue/test-utils";

describe("Copyright", () => {
  const { setLocale } = useLocaleStore(appConfig.slug);

  const locales = ["en", "fr"];

  for (const locale of locales) {
    setLocale(locale);

    const wrapper: VueWrapper = shallowMount(Copyright, {
      global: {
        plugins: [
          [
            i18nPlugin,
            {
              config: appConfig,
            },
          ],
        ],
      },
      props: {
        holder: appConfig.copyright?.holder as string,
        url: appConfig.copyright?.url as string,
      },
    });

    it("matches snapshot " + locale, () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  }
});
