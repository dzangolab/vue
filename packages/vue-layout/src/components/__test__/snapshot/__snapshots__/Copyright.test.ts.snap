// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Copyright > matches snapshot en 1`] = `
<div
  class="copyright"
>
   © 2023 
  <a
    href="https://dzango.com"
    target="_blank"
  >
    Dzango Technologies Limited
  </a>
    All rights reserved.
</div>
`;

exports[`Copyright > matches snapshot fr 1`] = `
<div
  class="copyright"
>
   © 2023 
  <a
    href="https://dzango.com"
    target="_blank"
  >
    Dzango Technologies Limited
  </a>
    Tous droits réservés.
</div>
`;
